goagen/codegen/types.go

```diff
@@ -113,7 +113,11 @@ func goTypeDefObject(obj design.Object, def *design.AttributeDefinition, tabs in
 		typedef := ""
 		if tname, ok := field.Metadata["struct:field:type"]; ok {
 			if len(tname) > 0 {
-				typedef = tname[0]
+				if (field.Type.IsPrimitive() && private) || field.Type.IsObject() || def.IsPrimitivePointer(name) {
```

Comment:
Might be good to remove the copy/paste here and define a variable below `typedef := ""` on line 113 that can be used both here and below. Something like:0
```go
pointer := ""
if (field.Type.IsPrimitive() && private) || field.Type.IsObject() || def.IsPrimitivePointer(name) {
    pointer = "*"
}
```
then this code becomes `typedef = pointer + tname[0]` and the code below (line 125) becomes `typedef = pointer + typedef`